cmd/minishift/cmd/start.go

```diff
@@ -127,6 +128,12 @@ func runStart(cmd *cobra.Command, args []string) {
 	setOcProxy()
 	setShellProxy()
 
+	if viper.IsSet("http-proxy") {
+		proxyUrl = viper.GetString(httpProxy)
```

Comment:
It is here were you want to parse the proxy settings and put them into the RegistrationParameters. The in the registration code, you can just apply them, if they are set.

If we do this together with the username and password we actually should do this in _root.go_. This would do the right thing for all command which potentially need these values. I think this would be the easier approach for now and then decide on issue #432